vllm/model_executor/layers/linear.py

```diff
@@ -775,3 +828,105 @@ def extra_repr(self) -> str:
         s += f", tp_size={self.tp_size}"
         s += f", reduce_results={self.reduce_results}"
         return s
+
+
+class FusedMoELinear(torch.nn.Module):
+
```

Comment:
It's worth putting a docstring here for model implementers clarifying what w1, w2, w3 usually map to and what structure this layer encapsulates for MoEs in vLLM

Reply:
Done